autogpt_platform/backend/backend/blocks/example/example.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import Any
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema, BlockType
+from backend.data.model import (
+    APIKeyCredentials,
+    ContributorDetails,
+    CredentialsField,
+    SchemaField,
+)
+
+from ._api import ExampleClient
+from ._auth import TEST_CREDENTIALS, TEST_CREDENTIALS_INPUT, ExampleCredentialsInput
+
+logger = logging.getLogger(__name__)
+
+
+class ExampleBlock(Block):
+
+    class Input(BlockSchema):
+        name: str = SchemaField(
+            description="The name of the example block", placeholder="Enter a name"
+        )
+        greetings: list[str] = SchemaField(
+            description="The greetings to display", default=["Hello", "Hi", "Hey"]
+        )
+        is_funny: bool = SchemaField(
+            description="Whether the block is funny", placeholder="True", default=True
+        )
+        # Only if the block needs credentials
+        credentials: ExampleCredentialsInput = CredentialsField(
+            description="The credentials for the example block"
+        )
+
+    class Output(BlockSchema):
+        response: dict[str, Any] = SchemaField(
+            description="The response object generated by the example block."
+        )
+        all_responses: list[dict[str, Any]] = SchemaField(
```

Comment:
all_response can confuse the user especially when described as `All the responses from the example block`.
I think this field only represents responses that are grouped within a single block execution.

Reply:
This was added based upon a suggestion by Nick that the best practices was to yield each element of a list and the entire list as well. 

`all_reponses` is the yielding of the complete list, whilst `response` is yielding of each element within the list

Reply:
i think moving to pydantic for this would make it clearer 
```
class Horse(BaseModel):
  blah: int

horse: Horse = SchemaField(...)
horses: list[Horse] = SchemaField(...)
```

Reply:
I've changed it to use pydantic objects, however this is an open discussion in discord regarding best practice for yielding a each element of the list and the list or just each element of the list